src/transport_ofi.h

```diff
@@ -1314,8 +1314,17 @@ void shmem_transport_atomic_fetch(shmem_transport_ctx_t* ctx, void *target,
                                   const void *source, size_t len, int pe,
                                   int datatype)
 {
+#ifdef ENABLE_MR_ENDPOINT
```

Comment:
It seems this restriction is not directly related to `FI_MR_ENDPOINT` right?  But, I cannot think of a good alternative right now.. I'm just making the note that his might be possible to resolve/cleanup later, perhaps in #1077.

Also, I'm a little confused about this comment - let me make sure I understand.  Fetching atomics always have `FI_DELIVERY_COMPLETE` semantics via `fi_fetch_atomic` ([stated here](https://ofiwg.github.io/libfabric/v1.14.1/man/fi_atomic.3.html)), right?  But because `FI_DELIVERY_COMPLETE` atomics are not always supported, we can use `FI_INJECT` through `fi_fetch_atomicmsg` in those cases - is that it?  If so, I suppose another approach is for the blocking fetching atomic to have a branch (on the critical path 👎) that uses `fi_fetch_atomicmsg` on providers like this.

I mention that approach because I'm also concerned that only calling `fetch_atomic_nbi` here could lead to premature access of the result buffer.  Don't we need to quiet the operation before the user possibly accesses the result?  (It almost seems like a really weird use-case for `shmem_internal_pe_quiet` ;).

Reply:
Yes... I am not sure what else to use for the `ifdef` check. May be, a more restricted check will be by adding `ENABLE_FI_MANUAL_PROGRESS`. But, that will look even more weird. But, that check will definitely ensure that this is done for cxi provider.

I understand your concern. We can add that branch in the blocking fetch atomic.. but my thinking is we only want this for CXI, other providers can use `fi_fetch_atomic` with `FI_DELIVERY_COMPLETE`. Also, hoping that CXI will support this some day, and we can get rid of this small detour. 

The `quiet` here is not required, because this is originated from `shmem_internal_atomic_fetch` call at atomic_c.c4 line 396. There is a `get_wait` call there right after it. That will ensure that the result location is not accessed before the operation is complete. Let me know if I got it wrong here.

Reply:
Oh right - I keep forgetting about the `get_wait` in the top layer (and I don't see any other internal fetching atomics) - so that should be fine!

I agree on the other concerns - I'll mark this one as resolved!